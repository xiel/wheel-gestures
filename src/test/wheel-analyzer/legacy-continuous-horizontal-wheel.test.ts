import { recordPhases } from '../helper/recordPhases'

import macChrome from '../fixtures/legacy-mouse/mac-chrome-continuous-horizontal-wheel-right.json'
import winFF from '../fixtures/legacy-mouse/win-FF-continuous-horizontal-wheel-right.json'
import winChrome from '../fixtures/legacy-mouse/win-Chrome-continuous-horizontal-wheel-right.json'
import winChromeNew from '../fixtures/legacy-mouse/win-chrome-new.json'
import winEdge from '../fixtures/legacy-mouse/win-Edge.json'

describe('continuous-horizontal-wheel-right', () => {
  it('mac - chrome', () => {
    expect(recordPhases(macChrome.wheelEvents)).toMatchSnapshot()
  })

  it('win - FF', () => {
    expect(recordPhases(winFF.wheelEvents)).toMatchSnapshot()
  })

  it('Windows - Chrome', () => {
    expect(recordPhases(winChrome.wheelEvents)).toMatchSnapshot()
  })

  it('Windows - Chrome New', () => {
    expect(recordPhases(winChromeNew.wheelEvents)).toMatchSnapshot()
  })

  it('Windows - Edge', () => {
    expect(recordPhases(winEdge.wheelEvents)).toMatchSnapshot()
  })
})
