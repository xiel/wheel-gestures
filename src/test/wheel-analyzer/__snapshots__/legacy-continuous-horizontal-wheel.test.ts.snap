// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`continuous-horizontal-wheel-right Windows - Chrome 1`] = `
Object {
  "phases": Array [
    Object {
      "lastData": Object {
        "axisDeltas": Array [
          -900,
          0,
        ],
        "axisVelocity": Array [
          -0.7494940915187284,
          0,
        ],
        "debugData": null,
        "deltaTotal": -900,
        "deltaVelocity": 0,
        "isMomentum": false,
        "type": "WHEEL_END",
        "willEndSoon": false,
      },
      "range": Array [
        0,
        5,
      ],
      "wheelType": "WHEEL",
    },
    Object {
      "lastData": Object {
        "axisDeltas": Array [
          -900,
          0,
        ],
        "axisVelocity": Array [
          -0.7494940915187284,
          0,
        ],
        "debugData": null,
        "deltaTotal": -900,
        "deltaVelocity": 0,
        "isMomentum": false,
        "type": "ANY_WHEEL_END",
        "willEndSoon": false,
      },
      "range": Array [
        0,
        5,
      ],
      "wheelType": "ANY_WHEEL",
    },
  ],
}
`;

exports[`continuous-horizontal-wheel-right Windows - Chrome New 1`] = `
Object {
  "phases": Array [
    Object {
      "lastData": Object {
        "axisDeltas": Array [
          -3300,
          0,
        ],
        "axisVelocity": Array [
          -0.7501406515262076,
          0,
        ],
        "debugData": null,
        "deltaTotal": -3300,
        "deltaVelocity": 0,
        "isMomentum": false,
        "type": "WHEEL_END",
        "willEndSoon": false,
      },
      "range": Array [
        0,
        21,
      ],
      "wheelType": "WHEEL",
    },
    Object {
      "lastData": Object {
        "axisDeltas": Array [
          -3300,
          0,
        ],
        "axisVelocity": Array [
          -0.7501406515262076,
          0,
        ],
        "debugData": null,
        "deltaTotal": -3300,
        "deltaVelocity": 0,
        "isMomentum": false,
        "type": "ANY_WHEEL_END",
        "willEndSoon": false,
      },
      "range": Array [
        0,
        21,
      ],
      "wheelType": "ANY_WHEEL",
    },
  ],
}
`;

exports[`continuous-horizontal-wheel-right Windows - Edge 1`] = `
Object {
  "phases": Array [
    Object {
      "lastData": Object {
        "axisDeltas": Array [
          -1050,
          0,
        ],
        "axisVelocity": Array [
          -1.6410647397072176,
          0,
        ],
        "debugData": null,
        "deltaTotal": -1050,
        "deltaVelocity": 0,
        "isMomentum": false,
        "type": "WHEEL_END",
        "willEndSoon": false,
      },
      "range": Array [
        0,
        6,
      ],
      "wheelType": "WHEEL",
    },
    Object {
      "lastData": Object {
        "axisDeltas": Array [
          -1050,
          0,
        ],
        "axisVelocity": Array [
          -1.6410647397072176,
          0,
        ],
        "debugData": null,
        "deltaTotal": -1050,
        "deltaVelocity": 0,
        "isMomentum": false,
        "type": "ANY_WHEEL_END",
        "willEndSoon": false,
      },
      "range": Array [
        0,
        6,
      ],
      "wheelType": "ANY_WHEEL",
    },
  ],
}
`;

exports[`continuous-horizontal-wheel-right mac - chrome 1`] = `
Object {
  "phases": Array [
    Object {
      "lastData": Object {
        "axisDeltas": Array [
          -2821.7535400390625,
          0,
        ],
        "axisVelocity": Array [
          -0.9577288117430209,
          0,
        ],
        "debugData": null,
        "deltaTotal": -2821.7535400390625,
        "deltaVelocity": 0,
        "isMomentum": false,
        "type": "WHEEL_END",
        "willEndSoon": false,
      },
      "range": Array [
        0,
        22,
      ],
      "wheelType": "WHEEL",
    },
    Object {
      "lastData": Object {
        "axisDeltas": Array [
          -2821.7535400390625,
          0,
        ],
        "axisVelocity": Array [
          -0.9577288117430209,
          0,
        ],
        "debugData": null,
        "deltaTotal": -2821.7535400390625,
        "deltaVelocity": 0,
        "isMomentum": false,
        "type": "ANY_WHEEL_END",
        "willEndSoon": false,
      },
      "range": Array [
        0,
        22,
      ],
      "wheelType": "ANY_WHEEL",
    },
  ],
}
`;

exports[`continuous-horizontal-wheel-right win - FF 1`] = `
Object {
  "phases": Array [
    Object {
      "lastData": Object {
        "axisDeltas": Array [
          -1350,
          0,
        ],
        "axisVelocity": Array [
          -0.27621483375959077,
          0,
        ],
        "debugData": null,
        "deltaTotal": -1350,
        "deltaVelocity": 0,
        "isMomentum": false,
        "type": "WHEEL_END",
        "willEndSoon": false,
      },
      "range": Array [
        0,
        24,
      ],
      "wheelType": "WHEEL",
    },
    Object {
      "lastData": Object {
        "axisDeltas": Array [
          -1350,
          0,
        ],
        "axisVelocity": Array [
          -0.27621483375959077,
          0,
        ],
        "debugData": null,
        "deltaTotal": -1350,
        "deltaVelocity": 0,
        "isMomentum": false,
        "type": "ANY_WHEEL_END",
        "willEndSoon": false,
      },
      "range": Array [
        0,
        24,
      ],
      "wheelType": "ANY_WHEEL",
    },
  ],
}
`;
